File: solutions/elixir/bob/1/lib/bob.ex
defmodule Bob do
  def hey(input) do
    cond do
      String.upcase( input ) == input && Regex.run( ~r/[A-Za-z]/u, input ) && String.last( input ) == "?" -> "Calm down, I know what I'm doing!"
      String.trim( input ) == "" -> "Fine. Be that way!"
      String.trim( input ) |> String.last == "?" -> "Sure."
      String.upcase( input ) == input && Regex.run( ~r/\p{L}/u, input ) -> "Whoa, chill out!"
      true -> "Whatever."
    end
  end
end

File: solutions/csharp/bob/1/Bob.cs
using System;

public static class Bob
{
    public static string Response(string statement) => statement.Trim() switch 
    {
        var stm when stm.ToUpper() == stm && stm.ToLower() != stm && stm.EndsWith("?") => "Calm down, I know what I'm doing!",
        var stm when stm.Length == 0 => "Fine. Be that way!",
        var stm when stm.EndsWith("?") => "Sure.", 
        var stm when stm.ToUpper() == stm && stm.ToLower() != stm => "Whoa, chill out!",
        _ => "Whatever.",
    };
}